doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -390,6 +390,49 @@ inputs.
 | `replica_id` | [`text`]    | The ID of a cluster replica. |
 | `hydrated`   | [`boolean`] | Whether the object is hydrated on the replica. |
 
+## `mz_index_advice`
+
+The `mz_index_advice` view provides recommendations on where to add (or remove) indexes and materialized view based on the interdependencies between objects.
+The recommendations are intended to reduce resource usage (CPU and memory) by creating indexes and materialized views to precompute intermediate results that can be reused across several objects.
+It also highlights indexes that can be removed and materialized views that can be turned into plain views when they are unnecessary.
+
+{{< callout >}}
+
+### Limitations and considerations
+
+The recommendations are based solely on object dependencies. They do not take
+the actual query definitions into account and as such, can be misleading and
+even increase resource usage. For example:
+
+- When a materialized view or an index has been created because they are queried
+  directly, that is not apparent from the object dependencies, and using plain
+  views (recommendation based solely on object dependencies) is inappropriate
+  for the actual query pattern.
+
+- When a view is depended on by multiple maintained objects that all use very
+  selective filters or use lots of projections that can be pushed into or even
+  beyond the view, adding an index (recommendation based soley on object
+  dependencies) may increase resource usage.
+
+- When indexes have been created to enable [delta
+  joins](/transform-data/optimization/#optimize-multi-way-joins-with-delta-joins),
+  removing indexes (recommendation based soley on object dependencies) may lead
+  to lower memory utilization but the join implementation can no longer choose a
+  delta join.
+
+As such, test to confirm that the proposed recommendations reduce resource usage
+for your actual usage patterns.
+
+{{< /callout >}}
+
+<!-- RELATION_SPEC mz_internal.mz_index_advice -->
+| Field            | Type        | Meaning  |
+| ---------------  | ----------- | -------- |
+| `object_id`      | [`text`]    | The ID of the object. Corresponds to [`mz_catalog.mz_indexes.id`](../mz_catalog#mz_indexes), [`mz_catalog.mz_materialized_views.id`](../mz_catalog#mz_materialized_views), [`mz_catalog.mz_views.id`](../mz_catalog#mz_views). |
+| `recommendation` | [`text`]    | A proposed action to either change the object (e.g. turn a materialized view into an index or vice versa) or keep the object unchanged. |
+| `details`        | [`text`]    | An explanation why the `recommendation` was proposed based on the dependencies of the object. |
+| `justification`  | [`list`]    | The object ids that support the `recommendation`. `details` resolves the ids to names, which can be abiguous in case of multiple schemas or databases. |
```

Comment:
typo

```suggestion
| `justification`  | [`list`]    | The object ids that support the `recommendation`. `details` resolves the ids to names, which can be ambiguous in case of multiple schemas or databases. |
```